classes/app/file/repository.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace assignsubmission_pxaiwriter\app\file;
+
+
+use assignsubmission_pxaiwriter\app\interfaces\factory as base_factory;
+use context;
+use Exception;
+use file_storage;
+use stored_file;
+
+/* @codeCoverageIgnoreStart */
+defined('MOODLE_INTERNAL') || die();
+/* @codeCoverageIgnoreEnd */
+
+class repository implements interfaces\repository
+{
+    private base_factory $factory;
+    private file_storage $storage;
+
+    public function __construct(base_factory $factory)
+    {
+        $this->factory = $factory;
+        $this->storage = $this->factory->moodle()->file_storage();
+    }
+
+    public function get_file_area(): string
+    {
+        if (defined('ASSIGNSUBMISSION_PXAIWRITER_FILEAREA'))
+        {
+            return ASSIGNSUBMISSION_PXAIWRITER_FILEAREA;
+        }
+        return 'submissions_pxaiwriter';
+    }
+
+    public function get_submission_files(context $context, object $submission): array
```

Comment:
You are only using the id?

```suggestion
    public function get_submission_files(context $context, int $submission_id): array
```